net/udpbroadcastrelay/main.c

```diff
@@ -0,0 +1,651 @@
+                if (ioctl(fd,SIOCGIFINDEX, &req) < 0) {
+                    perror("ioctl(SIOCGIFINDEX)");
+                    exit(1);
+                }
+                #ifdef __FreeBSD__
+                iface->ifindex = req.ifr_index;
+                #else
+                iface->ifindex = req.ifr_ifindex;
+                #endif
+            #endif
+        }
+
+        /* Request flags for this interface */
+        short ifFlags;
+        {
+            struct ifreq req;
+            memcpy(&req, &basereq, sizeof(req));
+            if (ioctl(fd,SIOCGIFFLAGS, &req) < 0) {
+                perror("ioctl(SIOCGIFFLAGS)");
+                exit(1);
+            }
+            ifFlags = req.ifr_flags;
+        }
+
+        /* if the interface is not up or a loopback, ignore it */
+        if ((ifFlags & IFF_UP) == 0 || (ifFlags & IFF_LOOPBACK)) {
+            continue;
+        }
+
+        /* Get local IP for interface */
+        {
+            struct ifreq req;
+            memcpy(&req, &basereq, sizeof(req));
+            if (ioctl(fd,SIOCGIFADDR, &req) < 0) {
+                perror("ioctl(SIOCGIFADDR)");
+                exit(1);
+            }
+            memcpy(
+                &iface->ifaddr,
+                &((struct sockaddr_in *)&req.ifr_addr)->sin_addr,
+                sizeof(struct in_addr)
+            );
+        }
+
+        /* Get broadcast address for interface */
+        {
+            struct ifreq req;
+            memcpy(&req, &basereq, sizeof(req));
+            if (ifFlags & IFF_BROADCAST) {
+                if (ioctl(fd,SIOCGIFBRDADDR, &req) < 0) {
+                    perror("ioctl(SIOCGIFBRDADDR)");
+                    exit(1);
+                }
+                memcpy(
+                    &iface->dstaddr,
+                    &((struct sockaddr_in *)&req.ifr_broadaddr)->sin_addr,
+                    sizeof(struct in_addr)
+                );
+            } else {
+                if (ioctl(fd,SIOCGIFDSTADDR, &req) < 0) {
+                    perror("ioctl(SIOCGIFBRDADDR)");
+                    exit(1);
+                }
+                memcpy(
+                    &iface->dstaddr,
+                    &((struct sockaddr_in *)&req.ifr_dstaddr)->sin_addr,
+                    sizeof(struct in_addr)
+                );
+            }
+        }
+
+        char ifaddr[255];
+        inet_ntoa2(iface->ifaddr, ifaddr, sizeof(ifaddr));
+        char dstaddr[255];
+        inet_ntoa2(iface->dstaddr, dstaddr, sizeof(dstaddr));
+
+        DPRINT(
+            "%s: %i / %s / %s\n",
+            basereq.ifr_name,
+            iface->ifindex,
+            ifaddr,
+            dstaddr
+        );
+
+        // Set up a one raw socket per interface for sending our packets through
+        if((iface->raw_socket = socket(AF_INET,SOCK_RAW,IPPROTO_RAW)) < 0) {
+            perror("socket");
+            exit(1);
+        }
+        {
+            int yes = 1;
+            int no = 0;

```

Comment:
C99 defines booleans in stdbool.h as 0 and 1, so you can use true and false if you want.

http://openbook.rheinwerk-verlag.de/c_von_a_bis_z/005_c_basisdatentypen_015.htm

Reply:
Done